File: atcoder/abc393a_poisonous_oyster.nim
# Vicfred
# https://atcoder.jp/contests/abc393/tasks/abc393_a
# implementation

import strutils

# Read a single line with two words
let input = stdin.readLine()

# Split input into two words
let words = input.split(" ")

if words.len == 2:
    let (s1, s2) = (words[0], words[1])
    # Check conditions and print corresponding values
    if s1 == "fine" and s2 == "fine":
        echo 4
    elif s1 == "fine" and s2 == "sick":
        echo 3
    elif s1 == "sick" and s2 == "fine":
        echo 2
    elif s1 == "sick" and s2 == "sick":
        echo 1
    else:
        echo "Invalid input"
else:
    echo "Invalid input"

File: atcoder/abc393a_poisonous_oyster.zig
// vicfred
// https://atcoder.jp/contests/abc393/tasks/abc393_a
// implementation

const std = @import("std");

pub fn main() !void {
    var buf: [100]u8 = undefined;
    var stdin = std.io.getStdIn().reader();
    
    // Read a single line with two words
    const input = try stdin.readUntilDelimiterOrEof(&buf, '\n');
    if (input == null) {
        std.debug.print("Invalid input\n", .{});
        return;
    }
    
    var it = std.mem.split(u8, input.?, " ");
    const s1 = it.next() orelse "";
    const s2 = it.next() orelse "";
    
    // Check conditions and print corresponding values
    if (std.mem.eql(u8, s1, "fine") and std.mem.eql(u8, s2, "fine")) {
        std.debug.print("4\n", .{});
    } else if (std.mem.eql(u8, s1, "fine") and std.mem.eql(u8, s2, "sick")) {
        std.debug.print("3\n", .{});
    } else if (std.mem.eql(u8, s1, "sick") and std.mem.eql(u8, s2, "fine")) {
        std.debug.print("2\n", .{});
    } else if (std.mem.eql(u8, s1, "sick") and std.mem.eql(u8, s2, "sick")) {
        std.debug.print("1\n", .{});
    } else {
        std.debug.print("Invalid input\n", .{});
    }
}
